### PR TITLE
Only require DataSource when persistent timers are present

### PR DESCRIPTION
**Pull request by aguibert**
Proposed fix for issue #5099

### CODE REVIEW

**Comment by aguibert** [comment]
#build

**Comment by tkburroughs** [comment]
So, the concern with this fix is that it ignores programmatically created persistent timers. If timers were created in the database in the past, server is re-started but database cannot be accessed, then those persistent timers will not run... and the application likely won't behave properly.

Unlike the 'automatic' (i.e. `@Schedule`) timers that are processed in EJBMDOrchestrator, the only way to determine if the application supports programmatic timers is to scan all of the bytecode for all classes in the application and look for calls to the EJBTimerService methods that programmatically create timers, etc, which really isn't practical.

So, with the proposed fix, let's say the app did create programmatic persistent timers in the past.  When the server is restarted.... they won't ever run, because the persistent timer service won't be started... unless the app also happened to have an `@Schedule` of a persistent timer.

I think a more appropriate change would be to detect when a persistent timer datasource just hasn't been configured at all, and if so, then just log a warning, not an error, and allow the application to start.

File: dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ws/metadata/ejb/EJBMDOrchestrator.java [hunk 1]
```diff
@@ -5011,7 +5015,10 @@ else if (!bmd.metadataComplete) {
 
             // F743-13022 - Check now if the runtime environment supports timers
             // rather than waiting until the automatic timers are created.
-            bmd.container.getEJBRuntime().setupTimers(bmd);
+            if (isTraceOn && tc.isDebugEnabled())
+                Tr.debug(tc, "Are persistent timers present? " + hasPersistentTimers);
+            if (hasPersistentTimers)
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/src/com/ibm/ws/metadata/ejb/EJBMDOrchestrator.java]: Note: if persistent timers exist in the database, they won't run if the server is restarted and there are no `@Schedule` persistent timers.

**Comment by aguibert** [comment]
thanks for the feedback @tkburroughs! That makes perfect sense about programmatic timers.

> I think a more appropriate change would be to detect when a persistent timer datasource just hasn't been configured at all, and if so, then just log a warning, not an error, and allow the application to start.

Sounds good to me.  I'll go ahead and cancel this PR in favor of the above solution.

**Pull request closed without merging**